src/gen/opcodes.rs

```diff
@@ -107,196 +111,56 @@ fn opcode_tester_with_assert_before(a: &mut Allocator, val: &[u8]) -> Option<Con
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_COIN_ANNOUNCEMENT]),
-        Some(ASSERT_COIN_ANNOUNCEMENT)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[CREATE_PUZZLE_ANNOUNCEMENT]),
-        Some(CREATE_PUZZLE_ANNOUNCEMENT)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_PUZZLE_ANNOUNCEMENT]),
-        Some(ASSERT_PUZZLE_ANNOUNCEMENT)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_MY_COIN_ID]),
-        Some(ASSERT_MY_COIN_ID)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_MY_PARENT_ID]),
-        Some(ASSERT_MY_PARENT_ID)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_MY_PUZZLEHASH]),
-        Some(ASSERT_MY_PUZZLEHASH)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_MY_AMOUNT]),
-        Some(ASSERT_MY_AMOUNT)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_SECONDS_RELATIVE]),
-        Some(ASSERT_SECONDS_RELATIVE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_SECONDS_ABSOLUTE]),
-        Some(ASSERT_SECONDS_ABSOLUTE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_HEIGHT_RELATIVE]),
-        Some(ASSERT_HEIGHT_RELATIVE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_HEIGHT_ABSOLUTE]),
-        Some(ASSERT_HEIGHT_ABSOLUTE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_BEFORE_SECONDS_RELATIVE]),
-        Some(ASSERT_BEFORE_SECONDS_RELATIVE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_BEFORE_SECONDS_ABSOLUTE]),
-        Some(ASSERT_BEFORE_SECONDS_ABSOLUTE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_BEFORE_HEIGHT_RELATIVE]),
-        Some(ASSERT_BEFORE_HEIGHT_RELATIVE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_BEFORE_HEIGHT_ABSOLUTE]),
-        Some(ASSERT_BEFORE_HEIGHT_ABSOLUTE)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_CONCURRENT_SPEND]),
-        Some(ASSERT_CONCURRENT_SPEND)
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_CONCURRENT_PUZZLE]),
-        Some(ASSERT_CONCURRENT_PUZZLE)
-    );
+#[cfg(test)]
+use rstest::rstest;
 
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ALWAYS_TRUE]),
-        Some(ALWAYS_TRUE)
-    );
-    // leading zeros are not allowed, it makes it a different value
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[ASSERT_HEIGHT_ABSOLUTE, 0]),
-        None
-    );
-    assert_eq!(
-        opcode_tester_with_assert_before(&mut a, &[0, ASSERT_HEIGHT_ABSOLUTE]),
-        None
-    );
-    assert_eq!(opcode_tester_with_assert_before(&mut a, &[0]), None);
+#[cfg(test)]
```

Comment:
Oh boy. I hope this is all the same meaning (I'm not that familiar with `rtest`, but it looks like you are!).

Reply:
I split out the parameterization into its own commit, to make it a bit cleaner at least.